model/jpa/src/main/java/org/keycloak/authorization/jpa/store/JPAResourceStore.java

```diff
@@ -145,15 +145,7 @@ private void findByOwnerFilter(String ownerId, String resourceServerId, Consumer
         }
 
         ResourceStore resourceStore = provider.getStoreFactory().getResourceStore();
-        List<String> result = query.getResultList();
-
-        for (String entity : result) {
-            Resource cached = resourceStore.findById(entity, resourceServerId);
-            
-            if (cached != null) {
-                consumer.accept(cached);
-            }
-        }
+        query.getResultStream().map(id -> resourceStore.findById(id.getId(), resourceServerId)).forEach(consumer);
```

Comment:
Not sure this is intentional, but currently the named query used here is not returning `id` but whole ResourceEntity, so I believe this should be possible:
```suggestion
        query.getResultStream().map(resourceEntity -> new ResourceAdapter(resourceEntity, entityManager, provider.getStoreFactory())).forEach(consumer);
```

Or if we want to make it possible to use cache for getting ResourceEntities, we should probably make the NamedQueries to return id only.

Reply:
It is intentional. If returning id, an additional query will happen later. That code is what I got better results.

Reply:
But now it is doing the same query twice. You get a list of ResrouceEntity (not only id but full ResourceEntity) and then for each ResrouceEntity you just get `id`, and then do another request for ResrouceEntity, which was already loaded by the previous query. 

You probably want to change `r` to `r.id` [here](https://github.com/keycloak/keycloak/pull/7227/files#diff-99e590ff1a5d4b89767173cfa4708f88R60) to not have two queries for the same ResrouceEntity. Or am I missing something?

Reply:
You should end up with two queries because the entity is already loaded into the entity manager. And we call the resourceStore.findById to force going through the cache layer.

Makes sense now ?